raiden/tests/integration/fixtures/raiden_network.py

```diff
@@ -154,6 +157,26 @@ def monitoring_service_contract_address() -> Address:
     return Address(bytes([1] * 20))
 
 
+@pytest.fixture
+def resolvers(resolver_ports) -> List[Optional[subprocess.Popen]]:
+    resolvers = []
+    for port in resolver_ports:
+        resolver = None
+        if port is not None:
+            args = ["python", "tools/dummy_resolver_server.py", str(port)]
+            resolver = subprocess.Popen(args, stdout=subprocess.PIPE)
+            assert resolver.poll() is None
+        resolvers.append(resolver)
+
+    return resolvers
```

Comment:
You can do a cool trick here to avoid having to manually cleanup the resolvers in each test they are used.

Make this into a generator fixture. Essentially change this line to:

```diff
-return resolvers
+yield resolvers
+
+stop_resolvers(resolvers)
```

You can also get rid of the `stop_resolvers()` function entirely and unroll it into the cleanup of this fixture.

We use it a lot in our tests. For example here:

https://github.com/raiden-network/raiden/blob/269e6652b151f3f377543de0662d8ad885b87247/raiden/tests/integration/fixtures/transport.py#L53-L87

Reply:
Done. Thanks!